src/util/temp-dir.js

```diff
@@ -76,7 +76,7 @@ export class TempDir {
       })
       .then(([tmpPath, removeTempDir]) => {
         this._path = tmpPath;
-        this._removeTempDir = removeTempDir;
+        this._removeTempDir = promisify(removeTempDir);
```

Comment:
This is incorrect. The "removeTempDir" variable here is [created by calling `_prepareTmpDirRemoveCallback`](https://github.com/raszi/node-tmp/blob/05aba23f06aa62dc286bca4340849f475f862dc3/lib/tmp.js#L333), defined at https://github.com/raszi/node-tmp/blob/05aba23f06aa62dc286bca4340849f475f862dc3/lib/tmp.js#L448-L463

Its return value is the result of calling `_prepareRemoveCallback`, which is defined at https://github.com/raszi/node-tmp/blob/05aba23f06aa62dc286bca4340849f475f862dc3/lib/tmp.js#L466-L477

This returns a function that takes 1 function parameter, called "next".

`util.promisify` uses Node.js's callback convention, and takes a function takes that the following parameters: `(error, result)`.

What you should be doing is to not call `promisify` here, but pass a callback to `this._removeTempDir()` below. That callback becomes the "next" function in the code that I referenced above.